### ISSUE TITLE
Is there existing implementations of learning on spatio-temporal graphs/ is it possible to make them using this framework?

### ISSUE THREAD

**Issue by jmy48** [rusty1s/pytorch_geometric#153]
## ❓ Questions & Help

Hi, I'm interested in learning over graphs that unroll through time via LSTM or other sequential learning mechanisms, like https://arxiv.org/pdf/1902.09130.pdf and https://arxiv.org/pdf/1803.07294.pdf. Is this possible/does this already exist?

**Comment by rusty1s** [comment]
Great question! I actually do have concrete plans to integrate (spatio-)temporal graph operators and datasets. For example, we already provide the [`BitconOTC`](https://rusty1s.github.io/pytorch_geometric/build/html/modules/datasets.html#torch_geometric.datasets.BitcoinOTC) dataset from the [EvolveGCN paper](https://arxiv.org/abs/1902.10191). IMO, more temporal graph datasets can be easily created by interpreting each timestamp as a single data object and using dataloaders without shuffling to evolve through time.
Concerning operators, I'm currently working on a [Know-Evolve](https://arxiv.org/abs/1705.05742) implementation, but I am very interested in adding other operators too. Feel free to submit a PR if you find the time to implement any of them.

**Comment by jmy48** [comment]
My use case is perfectly suited for know-evolve! Do you have an estimated completion time

**Comment by rusty1s** [comment]
It's done when it's done :P My guess is approximately 2 weeks. I am still learning about temporal point processes and their implementation and I have other stuff to do :(

**Comment by Lebo-Wang** [comment]
For my case, the new dataset with an extra "time" dimension is preferable. Keras layer of 'Timedistributed' could be applied for easy extension.

**Comment by rusty1s** [comment]
I have integrated the [RENet model](https://arxiv.org/pdf/1904.05530.pdf). See here for [documentation](https://rusty1s.github.io/pytorch_geometric/build/html/modules/nn.html#torch_geometric.nn.models.RENet) and an [example](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/renet.py).

**Comment by rusty1s** [comment]
```
ICEWS18-raw:
Epoch: 20, MRR: 0.2839, Hits@1: 0.1870, Hits@3: 0.3244, Hits@10: 0.4716

GDELT-raw:
Epoch: 20, MRR: 0.2082, Hits@1: 0.1321, Hits@3: 0.2248, Hits@10: 0.3555
```

**Comment by jacobharsten** [comment]
> 
> 
> I have integrated the [RENet model](https://arxiv.org/pdf/1904.05530.pdf). See here for [documentation](https://rusty1s.github.io/pytorch_geometric/build/html/modules/nn.html#torch_geometric.nn.models.RENet) and an [example](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/renet.py).

The documentation moved?

**Comment by rusty1s** [comment]
Yes, it moved to readthedocs. Have a look [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.models.RENet).
